source/extensions/http/stateful_session/cookie/cookie.h

```diff
@@ -38,6 +38,10 @@ class CookieBasedSessionStateFactory : public Envoy::Http::SessionStateFactory {
   CookieBasedSessionStateFactory(const CookieBasedSessionStateProto& config);
 
   Envoy::Http::SessionStatePtr create(const Envoy::Http::RequestHeaderMap& headers) const override {
+    if (bool request_path_match = requestPathMatch(headers.getPathValue()); !request_path_match) {
```

Comment:
nit: I think just `if (!requestPathMatch(headers.getPathValue()) {` would be better.

Reply:
get it.